test/e2e/tests/confirmations/transactions/contract-interaction-redesign.spec.js

```diff
@@ -0,0 +1,142 @@
+  const smartContract = SMART_CONTRACTS.PIGGYBANK;
+
+  if (!process.env.ENABLE_CONFIRMATION_REDESIGN) {
+    return;
+  }
+
+  it(`Opens a contract interaction type 0 transaction (Legacy)`, async function () {
+    await withFixtures(
+      {
+        dapp: true,
+        fixtures: new FixtureBuilder()
+          .withPermissionControllerConnectedToTestDapp()
+          .withPreferencesController({
+            preferences: { redesignedConfirmationsEnabled: true },
+          })
+          .build(),
+        ganacheOptions: defaultGanacheOptions,
+        smartContract,
+        title: this.test?.fullTitle(),
+      },
+      async ({ driver, contractRegistry }) => {
+        const contractAddress = await contractRegistry.getContractAddress(
+          smartContract,
+        );
+        await unlockWallet(driver);
+
+        await openDapp(driver, contractAddress);
+
+        await initiateContractInteractionTx(driver);
+      },
+    );
+  });
+
+  it(`Opens a contract interaction type 2 transaction (EIP1559)`, async function () {
+    await withFixtures(
+      {
+        dapp: true,
+        fixtures: new FixtureBuilder()
+          .withPermissionControllerConnectedToTestDapp()
+          .withPreferencesController({
+            preferences: { redesignedConfirmationsEnabled: true },
+          })
+          .build(),
+        ganacheOptions: defaultGanacheOptionsForType2Transactions,
+        smartContract,
+        title: this.test?.fullTitle(),
+      },
+      async ({ driver, contractRegistry }) => {
+        const contractAddress = await contractRegistry.getContractAddress(
+          smartContract,
+        );
+        await unlockWallet(driver);
+
+        await openDapp(driver, contractAddress);
+
+        await initiateContractInteractionTx(driver);
+      },
+    );
+  });
+
+  it(`Opens a contract interaction type 2 transaction that includes layer 1 fees breakdown on a layer 2`, async function () {
+    await withFixtures(
+      {
+        dapp: true,
+        fixtures: new FixtureBuilder({ inputChainId: CHAIN_IDS.OPTIMISM })
+          .withPermissionControllerConnectedToTestDapp()
+          .withPreferencesController({
+            preferences: { redesignedConfirmationsEnabled: true },
+          })
+          .withTransactionControllerOPLayer2Transaction()
+          .build(),
+        ganacheOptions: {
+          ...defaultGanacheOptionsForType2Transactions,
+          network_id: hexToNumber(CHAIN_IDS.OPTIMISM),
+          chainId: hexToNumber(CHAIN_IDS.OPTIMISM),
+        },
+        title: this.test?.fullTitle(),
+      },
+      async ({ driver }) => {
+        await unlockWallet(driver);
+
+        await openAdvancedDetailsAndCheckTheyExist(driver);
+      },
+    );
+  });
+});
+
+async function deployContract(driver) {
+  await driver.switchToWindowWithTitle(WINDOW_TITLES.TestDApp);
+  await driver.clickElement(`#deployButton`);
+
+  await driver.delay(2000);
```

Comment:
could we change the delay for a condition? If we expect for example 3 windows (MM + testdapp + dialog) we could do something like:
`await driver.waitUntilXWindowHandles(3);`
this way we make sure that the dialog is there and then switch to dialog